backend/routes/recommendationHelpers.js

```diff
@@ -0,0 +1,49 @@
+const calculateRecommendedServices = (allServices, userId) => {
+    // Assign weights
+    const DEFAULT_WEIGHT = 1;
+    const RATING_WEIGHT_4_5 = 3;
+    const FAVORITE_WEIGHT = 2;
+
+    // Calculate scores based on weights
+    const serviceScores = {};
+
+    allServices.forEach(service => {
+        serviceScores[service.id] = {
+            ...service,
+            score: DEFAULT_WEIGHT
+        };
+
+        // Add weight for favorited services
+        if (service.favoritedBy.some(fav => fav.id === userId)) {
+            console.log(`Service ${service.id} is favorited by user ${userId}`);
+            serviceScores[service.id].score += FAVORITE_WEIGHT;
+        }
+
+        // Add weight for highly rated services
+        const userReview = service.reviewsAndRatings.find(review => review.userId === userId);
+        if (userReview && userReview.rating >= 4) {
+            console.log(`Service ${service.id} is rated ${userReview.rating} by user ${userId}`);
```

Comment:
Remove console.log. You can probably search for `console.log` and find all callsites and remove all before submit the commit